data/lib/src/network/config/endpoint.dart

```diff
@@ -40,15 +40,21 @@ class Endpoint {
   static final ServicePath documents = ServicePath('/documents');
 
   static final ServicePath shares = ServicePath('/shares');
+
+  static final ServicePath sharedSpaces = ServicePath('/shared_spaces');
 }
 
 extension ServicePathExtension on ServicePath {
   String generateEndpointPath() {
     return '${Endpoint.rootPath}${path}';
   }
 
-  String generateEndpointPathWithParameter(String parameter) {
-    return '${Endpoint.rootPath}${path}/${parameter}';
+  ServicePath withQueryParameter(String queryParameter) {
+    return ServicePath('${path}?${queryParameter}');
```

Comment:
nice

Reply:
Forgot to add `&` for many query parameters :(

Reply:
WDYT ?
```
  ServicePath withQueryParameters(List<String> queryParameters) {
    return ServicePath('${path}?${queryParameters.join("&")}');
  }
```

Reply:
please test it.